src/chain/bitcoincorerpcchain.rs

```diff
@@ -23,19 +25,29 @@ impl ChainWatchInterface for BitcoinCoreRPCClientChain {
 		self.util.watch_all_txn()
 	}
 
-	fn broadcast_transaction(&self, _tx: &Transaction) {
-		unimplemented!()
-	}
-
 	fn register_listener(&self, listener: Weak<ChainListener>) {
 		self.util.register_listener(listener)
 	}
 }
 
+impl BroadcasterInterface for BitcoinCoreRPCClientChain {
+	fn broadcast_transaction(&self, tx: &Transaction) -> Result<(), Box<Error>> {
+		unimplemented!()
+	}
+}
+
 impl BitcoinCoreRPCClientChain {
 	pub fn new() -> BitcoinCoreRPCClientChain {
 		BitcoinCoreRPCClientChain {
 			util: ChainWatchInterfaceUtil::new(),
 		}
 	}
+
+	pub fn block_connected(&self, block: &Block, height: u32) {
```

Comment:
I kinda envisioned this would have some polling thread (or a function to hit to make it poll) pinging the RPC and then generate the events there, not expose a pub fn which lets users call block_connected?

Reply:
In case of SPV this is not polling, but an incoming headers message processed and will call this.